apps/web-client/src/lib/stores/inventory/table_content.ts

```diff
@@ -1,45 +1,85 @@
+	const fullTableRow = entity?.stream(ctx).entries.pipe(
+		switchMap((valueFromEntryStream) => {
+			// map entry to just isbns
+			const isbns = valueFromEntryStream.map((entry) => entry.isbn);
+
+			// return array of merged values of books and volume stock client
+			return from(db.getBooks(isbns)).pipe(
+				map((booksFromDb) => {
+					const booksMap = new Map();
+					booksFromDb.forEach((book) => book && booksMap.set(book.isbn, book));
+
+					// for each value from entry stream (volume stock client)
+					return valueFromEntryStream.map((individualEntry) => {
+						// find the corresponding book in booksFromDb
+						const bookData = booksMap.get(individualEntry.isbn);
+						// account for books not found
+						if (!bookData) return individualEntry;
+						// return bookData + entry value
+						const fullTableElement = { ...bookData, ...individualEntry };
+						return fullTableElement;
+					});
+				})
+			);
+		})
+	);
+
+	return fullTableRow;
+};
+
 /**
  * Creates a store that streams the entries to be displayed in the table, with respect to the content in the db and the current page (set by pagination element).
+ * @param db current context database
  * @param entity a note or warehouse interface
  * @param currentPageStore a store that containing the current page index
- * @param bookStore a store with book data - used to extend each entry from the entity store with the rest of the book data
+ * @param ctx debug context
  * @returns
  */
-export const createDisplayEntriesStore: CreateDisplayEntriesStore = (entity, currentPageStore, bookStore, ctx) => {
-	const entriesStore = readableFromStream(entity?.stream(ctx).entries, [], ctx);
+export const createDisplayEntriesStore: CreateDisplayEntriesStore = (db, entity, currentPageStore, ctx) => {
+	const fullTableRowStream = createDisplayRowStream(db, entity, ctx);
+	const displayRowStore = readableFromStream(fullTableRowStream, [], ctx);
+
 	// Create a derived store that streams the entries value from the content store
-	const displayEntries = derived(
-		[entriesStore, currentPageStore, bookStore],
-		([$entriesStore, $currentPageStore, $bookStore]) => {
-			debug.log(ctx, 'display_entries:derived:inputs')({ $entriesStore, $currentPageStore, $bookStore });
-			const start = $currentPageStore * 10;
-			const end = start + 10;
+	const displayEntries = derived([displayRowStore, currentPageStore], ([$displayRowStore, $currentPageStore]) => {
+		debug.log(ctx, 'display_entries:derived:inputs')({ $displayRowStore, $currentPageStore });
+		const start = $currentPageStore * 10;
+		const end = start + 10;
 
-			const res = $entriesStore.slice(start, end).map((entry) => ({
-				...$bookStore[entry.isbn],
-				...entry
-			}));
-			debug.log(ctx, 'display_entries:derived:res')(res);
-			return res;
-		}
-	);
+		const res = $displayRowStore.slice(start, end).map((entry) => ({
+			...entry
+		}));
+		debug.log(ctx, 'display_entries:derived:res')(res);
+		return res;
```

Comment:
Can just do `return $displayRowStore.slice(start, end)` (unnecessary map, also, no need for additional variable here: it's clear what we're returning)